src/test/kotlin/com/hibob/unittests/UserServiceTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.hibob.unittests
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.Mockito.*
+
+class UserServiceTest{
+
+    private val userDao = mock<UserDao>()
+    private val notificationService = mock<NotificationService>()
+    private val emailVerificationService = mock<EmailVerificationService>()
+    private val userService = UserService(userDao, notificationService, emailVerificationService)
+
+    @Test
+    fun `Registering a user who is registered already`() {
```

Comment:
You can improve the test names so that they will explain better what is expected to happen
For example:
`registerUser should throw exception if user already exists`

Reply:
I'll change the names to be more informative